### PR TITLE
fix: `mergeConfig` util function

### PR DESCRIPTION
**Pull request by xuchaobei**
# PR Details

Fix `mergeConfig` util function

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/cli/core/src/config/mergeConfig.ts [hunk 1]
```diff
@@ -61,13 +61,15 @@ export const mergeConfig = (
...
-      return [...target, ...source];
-    }
-    if (isFunction(source)) {
-      return Array.isArray(target)
-        ? [...target, source]
-        : [target, source].filter(Boolean);
+    if (Array.isArray(target)) {
+      if (Array.isArray(source)) {
+        return [...target, ...source];
+      } else {
```
Comment (chenjiahan) [inline packages/cli/core/src/config/mergeConfig.ts:67]: the else block is unnecessary

Comment (xuchaobei) [reply packages/cli/core/src/config/mergeConfig.ts:67]: 有必要吧，不然 68 行的逻辑是在哪里包含了呢？

Comment (chenjiahan) [reply packages/cli/core/src/config/mergeConfig.ts:67]: 我的意思是 else 本身不需要，不是里面的逻辑不需要哈，可以看下 ESLint 中的 [no-else-return](https://eslint.org/docs/rules/no-else-return) 这条规则

Comment (xuchaobei) [reply packages/cli/core/src/config/mergeConfig.ts:67]: 了解了，不过我们现在的lint规则里应该没有这条规则，不然 lint 应该就不过了

Comment (chenjiahan) [reply packages/cli/core/src/config/mergeConfig.ts:67]: 嗯嗯，考虑到 Lint 规则已经很多了，这条不加也没事 😂

**Review by caohuilin — APPROVED** [review]

**Pull request merged**